### PR TITLE
Try rolling a new version of googletest (2021->2023).

### PR DESCRIPTION
**Pull request by matanlurey**

### CODE REVIEW

**Comment by zanderso** [comment]
Looks like some updates to https://github.com/flutter/engine/blob/main/build/secondary/third_party/googletest/BUILD.gn are needed.

**Comment by matanlurey** [comment]
> Looks like some updates to https://github.com/flutter/engine/blob/main/build/secondary/third_party/googletest/BUILD.gn are needed.

Right, there is actually a simplified source setup now:

```gn
# Only add the "*-all.cc" files (and no headers) to improve maintainability
# from upstream refactoring. The "*-all.cc" files include the respective
# source files.
sources = [ "googletest/src/gtest-all.cc" ]
```

Running on CI to see how it feels :)

**Comment by matanlurey** [comment]
Looks like this will pass, with the exception of some breakages in Fuchsia I'm not sure how to resolve:

```txt
../../third_party/googletest/googlemock/include/gmock/gmock-matchers.h:4416:36: note: in instantiation of function template specialization 'testing::MatcherCast<const fuchsia::math::SizeU &, fuchsia::math::SizeU>' requested here
 4416 |           property_name, property, MatcherCast<const PropertyType&>(matcher)));
      |                                    ^
../../flutter/shell/platform/fuchsia/flutter/tests/external_view_embedder_unittests.cc:189:7: note: in instantiation of function template specialization 'testing::Property<fuchsia::ui::composition::ImageProperties, const fuchsia::math::SizeU &, fuchsia::math::SizeU>' requested here
  189 |       Property("size", &fuchsia::ui::composition::ImageProperties::size, size));
      |       ^
```

It might just mean gmock needs to get rolled too? I'll try that.

File: shell/platform/fuchsia/flutter/tests/fakes/scenic/fake_flatland_types.h [hunk 1]
```diff
@@ -56,6 +59,20 @@ inline bool operator==(const fuchsia::math::RectF& a,
...
+    return false;
+  }
+  if (!a.has_value()) {
+  }
+  return a.value() == b.value();
+}
+
+}  // namespace math
+
+namespace composition {
```
[the remark this answers is not in the document]
Comment (matanlurey) [reply shell/platform/fuchsia/flutter/tests/fakes/scenic/fake_flatland_types.h]: Done.

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**